File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_GetWorldTimeRequest.js
import { GetWorldTimeRequest } from './pdu_jstype_GetWorldTimeRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduGetWorldTimeRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { GetWorldTimeRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint64(src.session_id);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { GetWorldTimeRequest } dst
     * @returns { GetWorldTimeRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.session_id = reader.read_uint64();
        return dst;
    }

    /**
     * @param { GetWorldTimeRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { GetWorldTimeRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new GetWorldTimeRequest());
    }
}

File: pdu/python/hako_srv_msgs/pdu_cdr_conv_GetWorldTimeRequest.py
from .pdu_pytype_GetWorldTimeRequest import GetWorldTimeRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_GetWorldTimeRequest(writer: CdrWriter, src: GetWorldTimeRequest):
    writer.write_uint64(src.session_id)


def cdr_body_to_py_GetWorldTimeRequest(reader: CdrReader, dst: GetWorldTimeRequest):
    dst.session_id = reader.read_uint64()
    return dst


def py_to_cdr_GetWorldTimeRequest(src: GetWorldTimeRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_GetWorldTimeRequest(writer, src)
    return writer.bytes()


def cdr_to_py_GetWorldTimeRequest(cdr_payload) -> GetWorldTimeRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = GetWorldTimeRequest()
    return cdr_body_to_py_GetWorldTimeRequest(reader, dst)
